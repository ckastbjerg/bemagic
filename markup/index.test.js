/* eslint-env jest */
const index = require('./index');

test('returns markup for a complex component', () => {
    global.config = require('../config');

    expect(index({
        components: {},
        themes: [
            {
                name: 'secondary',
                classes: 'prefix-theme-secondary',
            },
            {
                name: 'inverse',
                classes: 'prefix-theme-inverse',
            },
        ],
        css: '.block { background: red; }',
    })).toMatchSnapshot();
});
