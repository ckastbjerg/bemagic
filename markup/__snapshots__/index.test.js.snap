// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns markup for a complex component 1`] = `
"
        <div
            class=\\"bemagic-app js-bemagic-app-template\\"
            data-theme-class=\\"theme\\"
            data-background-class=\\"background\\"
        >
            
            <style>.block { background: red; }</style>
            <div class=\\"bemagic-app__header bemagic-header js-header\\">
                <div class=\\"bemagic-header__logo js-header-item\\">BEMagic (ﾉ◕ヮ◕)ﾉ*:･ﾟ✧</div>
                
    <div class=\\"bemagic-header__item bemagic-header__themes js-header-item js-header-themes\\">
        Themes
        <div
            class=\\"bemagic-header__theme bemagic-theme-button js-theme-toggle is-active\\"
            data-class=\\"theme\\"
        ></div>
        
            <div
                class=\\"bemagic-header__theme bemagic-theme-button js-theme-toggle prefix-theme-secondary background\\"
                data-class=\\"prefix-theme-secondary\\"
            ></div>
        
            <div
                class=\\"bemagic-header__theme bemagic-theme-button js-theme-toggle prefix-theme-inverse background\\"
                data-class=\\"prefix-theme-inverse\\"
            ></div>
        
    </div>

                
    <div class=\\"bemagic-header__item bemagic-header__themes js-header-item js-header-backgrounds\\">
        undefined
    </div>

                
    

            </div>
            <div class=\\"bemagic-app__sidebar\\">
                
    <div class=\\"bemagic-menu js-menu\\">
        
    </div>

            </div>
            <div class=\\"bemagic-app__pages js-pages theme background\\">
                
            </div>
        </div>
    "
`;
